crates/bevy_utils/src/label.rs

```diff
@@ -5,9 +5,16 @@ use std::{
     hash::{Hash, Hasher},
 };
 
+/// An object safe version of [`Eq`]. This trait is automatically implemented
+/// for any `'static` type that implements `Eq`.
 pub trait DynEq: Any {
+    /// Casts the type to `dyn Any`.
     fn as_any(&self) -> &dyn Any;
 
+    /// This method tests for `self` and `other` values to be equal.
+    ///
+    /// Implementors should avoid returning `true` when the underlying types are
+    /// not the same.
```

Comment:
Nit: 

I think maybe you can be a little stronger with the language here. I don't say a reason at all for someone to return true if the underlying types are different. Though maybe I am missing a use case for something like that?

Maybe just replace `avoid` with `not`?

Reply:
That's an interesting one. Should `0u64.dyn_eq(0u32)` return true? IMO it probably should.

Reply:
Should this trait be manually implementable at all? Adding the bound `DynEq: Eq` should make any impls other than the blanket impl impossible.

Reply:
I'd rather not introduce a breaking change in a docs PR, and even then it may not be desirable force it to only work with same type to same type equality.

Reply:
Thats a good point!